app/src/test/java/mozilla/lockbox/presenter/FxALoginPresenterTest.kt

```diff
@@ -58,21 +72,36 @@ class FxALoginPresenterTest : DisposingTest() {
     val view = FakeFxALoginView(disposer)
 
     @Mock
-    val accountStore = PowerMockito.mock(AccountStore::class.java)
+    val accountStore = PowerMockito.mock(AccountStore::class.java)!!
 
     private val loginURLSubject = PublishSubject.create<String>()
+
+    val dispatcher = Dispatcher()
     private val dispatcherObserver = TestObserver.create<Action>()
 
+    @Mock
+    val networkStore = PowerMockito.mock(NetworkStore::class.java)!!
+
+    private var isConnected: Observable<Boolean> = PublishSubject.create()
+    var isConnectedObserver = TestObserver.create<Boolean>()
+
+    @Mock
+    private val connectivityManager = PowerMockito.mock(ConnectivityManager::class.java)
+
     lateinit var subject: FxALoginPresenter
 
     @Before
     fun setUp() {
+        whenCalled(networkStore.isConnected).thenReturn(isConnected)
         whenCalled(accountStore.loginURL).thenReturn(loginURLSubject)
 
         PowerMockito.whenNew(AccountStore::class.java).withAnyArguments().thenReturn(accountStore)
-        Dispatcher.shared.register.subscribe(dispatcherObserver)
+        dispatcher.register.subscribe(dispatcherObserver)
 
-        subject = FxALoginPresenter(view, accountStore = accountStore)
+        networkStore.connectivityManager = connectivityManager
```

Comment:
Not sure I understand this. The `networkStore` is a mock, and you're giving it mock `connectivityManager`?

Reply:
you can't create an instance of a ConnectivityManager, it has to be mocked. And the network store's connectivity manager has to be instantiated or it will break